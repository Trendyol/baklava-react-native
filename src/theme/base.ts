export const space = [0, 2, 4, 8, 12, 16, 24, 32, 48, 64];

export const fonts = ['System', 'System', 'System', 'System', 'System'];

export const fontSizes = [12, 14, 16, 20, 24, 28, 32];

export const fontWeights = ['300', '400', '500', '600', '700'];

export const lineHeights = [12, 14, 16, 20, 24, 28, 32, 36];

export const radii = {
  none: 0,
  small: 4,
  normal: 6,
  large: 8,
  xlarge: 16,
};

export const colors = {
  white: '#ffffff',
  black: '#000000',
  transparent: 'transparent',
  // Color
  primaryColor: '#F27A1A',
  secondaryColor: '#273142',
  tertiaryColor: '#F1F2F7',
  successColor: '#0BC15C',
  dangerColor: '#FF5043',
  warningColor: '#FFB600',
  alternativeColor: '#5794FF',
  featuredColor: '#8C4EFF',
  // Hover
  primaryHover: '#EF6114',
  secondaryHover: '#0F131A',
  tertiaryHover: '#D5D9E1',
  successHover: '#09A44E',
  dangerHover: '#FF3028',
  warningHover: '#FF9800',
  alternativeHover: '#457EFF',
  featuredHover: '#753EFF',
  // Background
  primaryBackground: '#F1F2F7',
  accentBackground: '#FEF2E8',
  secondaryBackground: '#FFFFFF',
  tertiaryBackground: '#F7F7FA',
  successBackground: '#E7F9EF',
  alertBackground: '#FFEEEC',
  warningBackground: '#FFF8E6',
  infoBackground: '#EEF4FF',
  featuredCBackground: '#F4EDFF',
  // Border Color
  borderColor: '#D5D9E1',
  // Content
  contentPrimary: '#273142',
  contentSecondary: '#6E7787',
  contentTertiary: '#95A1B5',
  contentPassive: '#AFBBCA',
};

export default {
  space,
  fonts,
  fontSizes,
  fontWeights,
  lineHeights,
  radii,
  colors,
};
