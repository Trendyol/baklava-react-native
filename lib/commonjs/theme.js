"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.space = exports.radii = exports.fontSizes = exports.default = exports.colors = void 0;
const space = [0, 2, 4, 8, 12, 16, 24, 32, 48, 64];
exports.space = space;
const fontSizes = [12, 14, 16, 20, 24, 28, 32];
exports.fontSizes = fontSizes;
const radii = {
  none: 0,
  small: 4,
  normal: 6,
  large: 8,
  xlarge: 16
};
exports.radii = radii;
const colors = {
  white: '#ffffff',
  black: '#000000',
  transparent: 'transparent',
  // Color
  primaryColor: '#F27A1A',
  secondaryColor: '#273142',
  tertiaryColor: '#F1F2F7',
  successColor: '#0BC15C',
  dangerColor: '#FF5043',
  warningColor: '#FFB600',
  alternativeColor: '#5794FF',
  featuredColor: '#8C4EFF',
  // Hover
  primaryHover: '#EF6114',
  secondaryHover: '#0F131A',
  tertiaryHover: '#D5D9E1',
  successHover: '#09A44E',
  dangerHover: '#FF3028',
  warningHover: '#FF9800',
  alternativeHover: '#457EFF',
  featuredHover: '#753EFF',
  // Background
  primaryBackground: '#F1F2F7',
  accentBackground: '#FEF2E8',
  secondaryBackground: '#FFFFFF',
  tertiaryBackground: '#F7F7FA',
  successBackground: '#E7F9EF',
  alertBackground: '#FFEEEC',
  warningBackground: '#FFF8E6',
  infoBackground: '#EEF4FF',
  featuredCBackground: '#F4EDFF',
  // Border Color
  borderColor: '#D5D9E1',
  // Content
  contentPrimary: '#273142',
  contentSecondary: '#6E7787',
  contentTertiary: '#95A1B5',
  contentPassive: '#AFBBCA',
  // Gradient
  gradientPurple: ['#8C4EFF', '#5794FF'],
  gradientOrange: ['#FFA964', '#F27A1A']
};
exports.colors = colors;
var _default = {
  space,
  fontSizes,
  radii,
  colors
};
exports.default = _default;
//# sourceMappingURL=theme.js.map