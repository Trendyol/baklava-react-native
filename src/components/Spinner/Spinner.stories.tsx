import { ComponentStory } from '@storybook/react-native';
import React from 'react';
import Text from '../Text/Text';
import Spinner from './Spinner';
import Box from '../Box/Box';

export default {
  title: 'Spinner',
  component: Spinner,
};

type SpinnerStory = ComponentStory<typeof Spinner>;

export const Size: SpinnerStory = () => (
  <>
    <Text p="2xs" variant="subtitle01Bold">
      Spinner
    </Text>
    <Box
      flexDirection="row"
      alignItems="flex-end"
      justifyContent="space-around">
      <Box>
        <Spinner size="2xl" />
        <Text mt="s" textAlign="center">
          2xl
        </Text>
      </Box>
      <Box>
        <Spinner size="xl" />
        <Text mt="s" textAlign="center">
          xl
        </Text>
      </Box>
      <Box>
        <Spinner size="l" />
        <Text mt="s" textAlign="center">
          l
        </Text>
      </Box>
      <Box>
        <Spinner size="m" />
        <Text mt="s" textAlign="center">
          m
        </Text>
      </Box>
      <Box>
        <Spinner size="s" />
        <Text mt="s" textAlign="center">
          s
        </Text>
      </Box>
      <Box>
        <Spinner size="xs" />
        <Text mt="s" textAlign="center">
          xs
        </Text>
      </Box>
      <Box>
        <Spinner size="2xs" />
        <Text mt="s" textAlign="center">
          2xs
        </Text>
      </Box>
    </Box>
  </>
);

export const Color: SpinnerStory = () => (
  <>
    <Text p="2xs" variant="subtitle01Bold">
      Spinner with color
    </Text>
    <Box flexDirection="row">
      <Box flex={1} p="m" justifyContent="space-around">
        <Box flexDirection="row" alignItems="center" mb="s">
          <Spinner size="l" color="neutralDarker" />
          <Text ml="m">neutralDarker</Text>
        </Box>
        <Box flexDirection="row" alignItems="center" mb="s">
          <Spinner size="l" color="red" />
          <Text ml="m">red</Text>
        </Box>
        <Box flexDirection="row" alignItems="center" mb="s">
          <Spinner size="l" color="maroon" />
          <Text ml="m">maroon</Text>
        </Box>
        <Box flexDirection="row" alignItems="center" mb="s">
          <Spinner size="l" color="#EBEF95" />
          <Text ml="m">#EBEF95</Text>
        </Box>
        <Box flexDirection="row" alignItems="center" mb="s">
          <Spinner size="l" color="#BEADFA" />
          <Text ml="m">#BEADFA</Text>
        </Box>
        <Box flexDirection="row" alignItems="center" mb="s">
          <Spinner size="l" color="#5C8984" />
          <Text ml="m">#5C8984</Text>
        </Box>
      </Box>

      <Box flex={1} p="m" justifyContent="space-around" bg="neutralDarker">
        <Box flexDirection="row" alignItems="center" mb="s">
          <Spinner size="l" color="neutralLight" />
          <Text ml="m" color="white">
            neutralLight
          </Text>
        </Box>
        <Box flexDirection="row" alignItems="center" mb="s">
          <Spinner size="l" color="red" />
          <Text ml="m" color="white">
            red
          </Text>
        </Box>
        <Box flexDirection="row" alignItems="center" mb="s">
          <Spinner size="l" color="maroon" />
          <Text ml="m" color="white">
            maroon
          </Text>
        </Box>
        <Box flexDirection="row" alignItems="center" mb="s">
          <Spinner size="l" color="#EBEF95" />
          <Text ml="m" color="white">
            #EBEF95
          </Text>
        </Box>
        <Box flexDirection="row" alignItems="center" mb="s">
          <Spinner size="l" color="#BEADFA" />
          <Text ml="m" color="white">
            #BEADFA
          </Text>
        </Box>
        <Box flexDirection="row" alignItems="center" mb="s">
          <Spinner size="l" color="#5C8984" />
          <Text ml="m" color="white">
            #5C8984
          </Text>
        </Box>
      </Box>
    </Box>
  </>
);
