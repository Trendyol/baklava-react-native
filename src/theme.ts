import { createTheme } from '@shopify/restyle';

const theme = createTheme({
  colors: {
    white: '#ffffff',
    black: '#000000',
    transparent: 'transparent',
    // Color
    primaryColor: '#F27A1A',
    secondaryColor: '#273142',
    tertiaryColor: '#F1F2F7',
    successColor: '#0BC15C',
    dangerColor: '#FF5043',
    warningColor: '#FFB600',
    alternativeColor: '#5794FF',
    featuredColor: '#8C4EFF',
    // Hover
    primaryHover: '#EF6114',
    secondaryHover: '#0F131A',
    tertiaryHover: '#D5D9E1',
    successHover: '#09A44E',
    dangerHover: '#FF3028',
    warningHover: '#FF9800',
    alternativeHover: '#457EFF',
    featuredHover: '#753EFF',
    // Background
    primaryBackground: '#F1F2F7',
    accentBackground: '#FEF2E8',
    secondaryBackground: '#FFFFFF',
    tertiaryBackground: '#F7F7FA',
    successBackground: '#E7F9EF',
    alertBackground: '#FFEEEC',
    warningBackground: '#FFF8E6',
    infoBackground: '#EEF4FF',
    featuredCBackground: '#F4EDFF',
    // Border Color
    borderColor: '#D5D9E1',
    // Content
    contentPrimary: '#273142',
    contentSecondary: '#6E7787',
    contentTertiary: '#95A1B5',
    contentPassive: '#AFBBCA',
  },
  spacing: {
    none: 0,
    '4xs': 2,
    '3xs': 4,
    '2xs': 8,
    xs: 12,
    s: 14,
    m: 16,
    l: 20,
    xl: 24,
    '2xl': 32,
    '3xl': 40,
    '4xl': 48,
    '5xl': 64,
    '6xl': 80,
  },
  borderRadii: {
    none: 0,
    xs: 2,
    s: 4,
    m: 6,
    l: 8,
    xl: 16,
  },
  breakpoints: {
    phone: 0,
    tablet: 768,
  },
  fonts: {
    light: 'System',
    regular: 'System',
    medium: 'System',
    semiBold: 'System',
    bold: 'System',
  },
  textVariants: {
    defaults: {
      fontSize: 14,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 16,
      color: 'contentPrimary',
    },
    // Heading
    heading1: {
      fontSize: 32,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 36,
    },
    heading2: {
      fontSize: 28,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 32,
    },
    heading3: {
      fontSize: 24,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 28,
    },
    // Subtitle01
    subtitle01Regular: {
      fontSize: 20,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 24,
    },
    subtitle01Medium: {
      fontSize: 20,
      fontWeight: '500',
      fontFamily: 'medium',
      lineHeight: 24,
    },
    subtitle01Semibold: {
      fontSize: 20,
      fontWeight: '600',
      fontFamily: 'semiBold',
      lineHeight: 24,
    },
    subtitle01Bold: {
      fontSize: 20,
      fontWeight: '700',
      fontFamily: 'bold',
      lineHeight: 24,
    },
    // Subtitle02
    subtitle02Regular: {
      fontSize: 16,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 20,
    },
    subtitle02Medium: {
      fontSize: 16,
      fontWeight: '500',
      fontFamily: 'medium',
      lineHeight: 20,
    },
    subtitle02Semibold: {
      fontSize: 16,
      fontWeight: '600',
      fontFamily: 'semiBold',
      lineHeight: 20,
    },
    subtitle02Bold: {
      fontSize: 16,
      fontWeight: '700',
      fontFamily: 'bold',
      lineHeight: 20,
    },
    // Subtitle03
    subtitle03Regular: {
      fontSize: 14,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 16,
    },
    subtitle03Medium: {
      fontSize: 14,
      fontWeight: '500',
      fontFamily: 'medium',
      lineHeight: 16,
    },
    subtitle03Semibold: {
      fontSize: 14,
      fontWeight: '600',
      fontFamily: 'semiBold',
      lineHeight: 16,
    },
    subtitle03Bold: {
      fontSize: 14,
      fontWeight: '700',
      fontFamily: 'bold',
      lineHeight: 16,
    },
    // Subtitle04
    subtitle04Regular: {
      fontSize: 12,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 14,
    },
    subtitle04Medium: {
      fontSize: 12,
      fontWeight: '500',
      fontFamily: 'medium',
      lineHeight: 14,
    },
    subtitle04Semibold: {
      fontSize: 12,
      fontWeight: '600',
      fontFamily: 'semiBold',
      lineHeight: 14,
    },
    subtitle04Bold: {
      fontSize: 12,
      fontWeight: '700',
      fontFamily: 'bold',
      lineHeight: 14,
    },
    // Body
    bodyText: {
      fontSize: 14,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 16,
    },
    bodyUnderline: {
      fontSize: 14,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 16,
      textDecorationLine: 'underline',
    },
    bodyTextLink: {
      fontSize: 14,
      fontWeight: '500',
      fontFamily: 'medium',
      lineHeight: 16,
      textDecorationLine: 'underline',
    },
    bodyLongText: {
      fontSize: 14,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 16,
    },
    // Caption
    captionText: {
      fontSize: 12,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 12,
    },
    captionMedium: {
      fontSize: 12,
      fontWeight: '500',
      fontFamily: 'medium',
      lineHeight: 12,
    },
    captionLongText: {
      fontSize: 12,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 12,
    },
    captionTextLink: {
      fontSize: 12,
      fontWeight: '500',
      fontFamily: 'medium',
      lineHeight: 12,
      textDecorationLine: 'underline',
    },
  },
});

export type Theme = typeof theme;
export default theme;
