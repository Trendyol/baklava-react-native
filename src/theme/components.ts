import { variant } from 'styled-system';
import {
  ButtonTypeTypes,
  ButtonVariantTypes,
} from '../components/Button/types';

// TODO: Unitless shorthands https://github.com/styled-components/css-to-react-native/issues/40
export const textStyle = {
  // Heading
  heading1: {
    fontSize: 6,
    fontWeight: 1,
    fontFamily: 1,
    // lineHeight: 7,
  },
  heading2: {
    fontSize: 5,
    fontWeight: 1,
    fontFamily: 1,
    // lineHeight: 6,
  },
  heading3: {
    fontSize: 4,
    fontWeight: 1,
    fontFamily: 1,
    // lineHeight: 5,
  },
  // Subtitle01
  subtitle01Regular: {
    fontSize: 3,
    fontWeight: 1,
    fontFamily: 1,
    // lineHeight: 4,
  },
  subtitle01Medium: {
    fontSize: 3,
    fontWeight: 2,
    fontFamily: 2,
    // lineHeight: 4,
  },
  subtitle01Semibold: {
    fontSize: 3,
    fontWeight: 3,
    fontFamily: 3,
    // lineHeight: 4,
  },
  subtitle01Bold: {
    fontSize: 3,
    fontWeight: 4,
    fontFamily: 4,
    // lineHeight: 4,
  },
  // Subtitle02
  subtitle02Regular: {
    fontSize: 2,
    fontWeight: 1,
    fontFamily: 1,
    // lineHeight: 3,
  },
  subtitle02Medium: {
    fontSize: 2,
    fontWeight: 2,
    fontFamily: 2,
    // lineHeight: 3,
  },
  subtitle02Semibold: {
    fontSize: 2,
    fontWeight: 3,
    fontFamily: 3,
    // lineHeight: 3,
  },
  subtitle02Bold: {
    fontSize: 2,
    fontWeight: 4,
    fontFamily: 4,
    // lineHeight: 3,
  },
  // Subtitle03
  subtitle03Regular: {
    fontSize: 1,
    fontWeight: 1,
    fontFamily: 1,
    // lineHeight: 2,
  },
  subtitle03Medium: {
    fontSize: 1,
    fontWeight: 2,
    fontFamily: 2,
    // lineHeight: 2,
  },
  subtitle03Semibold: {
    fontSize: 1,
    fontWeight: 3,
    fontFamily: 3,
    // lineHeight: 2,
  },
  subtitle03Bold: {
    fontSize: 1,
    fontWeight: 4,
    fontFamily: 4,
    // lineHeight: 2,
  },
  // Subtitle04
  subtitle04Regular: {
    fontSize: 0,
    fontWeight: 1,
    fontFamily: 1,
    // lineHeight: 1,
  },
  subtitle04Medium: {
    fontSize: 0,
    fontWeight: 2,
    fontFamily: 2,
    // lineHeight: 1,
  },
  subtitle04Semibold: {
    fontSize: 0,
    fontWeight: 3,
    fontFamily: 3,
    // lineHeight: 1,
  },
  subtitle04Bold: {
    fontSize: 0,
    fontWeight: 4,
    fontFamily: 4,
    // lineHeight: 1,
  },
  // Body
  bodyText: {
    fontSize: 1,
    fontWeight: 1,
    fontFamily: 1,
    // lineHeight: 2,
  },
  bodyUnderline: {
    fontSize: 1,
    fontWeight: 1,
    fontFamily: 1,
    // lineHeight: 2,
    textDecorationLine: 'underline',
  },
  bodyTextLink: {
    fontSize: 1,
    fontWeight: 2,
    fontFamily: 2,
    // lineHeight: 2,
    textDecorationLine: 'underline',
  },
  bodyLongText: {
    fontSize: 1,
    fontWeight: 1,
    fontFamily: 1,
    // lineHeight: 2,
  },
  // Caption
  captionText: {
    fontSize: 0,
    fontWeight: 1,
    fontFamily: 1,
    // lineHeight: 0,
  },
  captionMedium: {
    fontSize: 0,
    fontWeight: 2,
    fontFamily: 2,
    // lineHeight: 0,
  },
  captionLongText: {
    fontSize: 0,
    fontWeight: 1,
    fontFamily: 1,
    // lineHeight: 0,
  },
  captionTextLink: {
    fontSize: 0,
    fontWeight: 2,
    fontFamily: 2,
    // lineHeight: 0,
    textDecorationLine: 'underline',
  },
};

const buttonCommon = {
  borderRadius: 'normal',
};

export const buttonStyles = {
  height: [32, 40, 48],
  padding: [8, 12, 14, 16, 24, 32],
  variant: {
    primary: {
      color: 'white',
      backgroundColor: 'primaryColor',
      ...buttonCommon,
    },
    secondary: {
      color: 'white',
      backgroundColor: 'secondaryColor',
      ...buttonCommon,
    },
    success: {
      color: 'white',
      backgroundColor: 'successColor',
      ...buttonCommon,
    },
    danger: {
      color: 'white',
      backgroundColor: 'dangerColor',
      ...buttonCommon,
    },
  },
  type: (props: {
    variant: ButtonVariantTypes;
    isPressed: boolean;
    disabled: boolean;
    type: ButtonTypeTypes;
  }) => {
    const variantColor = props.disabled
      ? 'tertiaryColor'
      : `${props.variant}Color` ?? 'primaryBackground';
    const variantHoverColor = `${props.variant}Hover` ?? 'primaryBackground';

    return {
      contained: {
        color: props.disabled ? 'secondaryColor' : 'white',
        backgroundColor: props.disabled
          ? 'tertiaryColor'
          : props.isPressed
          ? variantHoverColor
          : variantColor,
        borderWidth: 1,
        borderColor: variantColor,
        ...buttonCommon,
      },
      outline: {
        color: props.isPressed ? 'white' : variantColor,
        backgroundColor: props.disabled
          ? variantColor
          : props.isPressed
          ? variantHoverColor
          : 'transparent',
        borderWidth: 1,
        borderColor: variantColor,
        ...buttonCommon,
      },
      text: {
        color: props.isPressed ? variantHoverColor : variantColor,
        backgroundColor: 'transparent',
        borderWidth: 1,
        borderColor: 'transparent',
        textDecorationLine: 'underline',
        ...buttonCommon,
      },
    };
  },
  size: (props: { text: string }) => {
    return {
      large: {
        paddingVertical: buttonStyles.padding[2],
        paddingHorizontal: props.text
          ? buttonStyles.padding[5]
          : buttonStyles.padding[2],
      },
      medium: {
        paddingVertical: buttonStyles.padding[1],
        paddingHorizontal: props.text
          ? buttonStyles.padding[4]
          : buttonStyles.padding[1],
      },
      small: {
        paddingVertical: buttonStyles.padding[0],
        paddingHorizontal: props.text
          ? buttonStyles.padding[3]
          : buttonStyles.padding[0],
      },
    };
  },
};

export const buttonVariant = variant({
  prop: 'variant',
  variants: buttonStyles.variant,
});

export const buttonType = (props: any) =>
  variant({
    prop: 'type',
    variants: buttonStyles.type(props),
  });

export const buttonSize = (props: any) =>
  variant({
    prop: 'size',
    variants: buttonStyles.size(props),
  });

export const inputStyle = {
  height: [32, 40, 48],
};

export const iconSizes = [12, 14, 16, 20, 24, 32];

export default {
  textStyle,
  buttonStyles,
  buttonVariant,
  buttonType,
  buttonSize,
  inputStyle,
  iconSizes,
};
