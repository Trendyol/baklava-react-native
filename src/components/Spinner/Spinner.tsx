import { VariantProps } from '@ergenekonyigit/restyle';
import React from 'react';
import {
  Animated,
  ColorValue,
  Easing,
  TextProps as RNTextProps,
} from 'react-native';
import Svg, { Path } from 'react-native-svg';
import theme, { Theme } from '../../theme';

const AnimatedSvg = Animated.createAnimatedComponent(Svg);

const Spinner = ({
  testID,
  accessibilityLabel,
  accessible,
  color = 'primaryKey',
  size = 'm',
}: {
  testID?: string;
  accessibilityLabel?: string;
  accessible?: boolean;
  color?: VariantProps<Theme, 'colors'>['variant'] | ColorValue;
  size?: VariantProps<Theme, 'spinnerSizeVariants'>['variant'];
}) => {
  const rotation = React.useRef(new Animated.Value(0)).current;
  const spinnerColor = theme.colors[color as keyof Theme['colors']] ?? color;
  const spinnerSize = theme.spinnerSizeVariants[size];

  React.useEffect(() => {
    const loop = Animated.loop(
      Animated.timing(rotation, {
        toValue: 1,
        duration: 1000,
        easing: Easing.linear,
        useNativeDriver: true,
      }),
    );
    loop.start();

    return () => {
      loop.stop();
    };
  }, [rotation]);

  const testProps = React.useMemo(() => {
    const result: RNTextProps = {
      accessible: false,
      accessibilityLabel: undefined,
      testID: undefined,
    };

    if (!(testID || accessibilityLabel)) {
      return result;
    }

    result.accessible = accessible ?? true;
    result.accessibilityLabel = accessibilityLabel ?? testID;
    result.testID = testID ?? accessibilityLabel;

    return result;
  }, [testID, accessibilityLabel, accessible]);

  const animatedStyle = React.useMemo(() => {
    return {
      transform: [
        {
          rotate: rotation.interpolate({
            inputRange: [0, 1],
            outputRange: ['0deg', '360deg'],
          }),
        },
      ],
    };
  }, [rotation]);

  return (
    <AnimatedSvg
      style={animatedStyle}
      width={spinnerSize}
      height={spinnerSize}
      viewBox="0 0 48 48"
      fill="#FFFFFF"
      xmlns="http://www.w3.org/2000/svg"
      {...testProps}>
      <Path
        opacity={0.3}
        d="M48 24c0 13.255-10.745 24-24 24S0 37.255 0 24 10.745 0 24 0s24 10.745 24 24zM7.2 24c0 9.278 7.522 16.8 16.8 16.8S40.8 33.278 40.8 24 33.278 7.2 24 7.2 7.2 14.722 7.2 24z"
        fill={spinnerColor}
      />
      <Path
        d="M44.4 24c1.988 0 3.628-1.62 3.33-3.587A23.999 23.999 0 0027.587.27C25.62-.028 24 1.612 24 3.6s1.63 3.561 3.573 3.984a16.8 16.8 0 0112.843 12.843C40.839 22.37 42.412 24 44.4 24z"
        fill={spinnerColor}
      />
    </AnimatedSvg>
  );
};

export default Spinner;
