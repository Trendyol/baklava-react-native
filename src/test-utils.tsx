import { ThemeProvider } from '@ergenekonyigit/restyle';
import { render } from '@testing-library/react-native';
import * as React from 'react';
import {
  GestureResponderHandlers,
  PanResponder,
  PanResponderCallbacks,
} from 'react-native';
import theme from './theme';
import { PortalProvider } from '.';

theme.fonts = {
  light: 'Rubik-Light',
  regular: 'Rubik-Regular',
  medium: 'Rubik-Medium',
  semiBold: 'Rubik-SemiBold',
  bold: 'Rubik-Bold',
};

export interface RenderOptions {
  wrapper?: React.ComponentType<any>;
  createNodeMock?: (element: React.ReactElement<any>) => any;
}

const AllProviders = ({ children }: { children?: React.ReactNode }) => (
  <ThemeProvider theme={theme}>
    <PortalProvider>{children}</PortalProvider>
  </ThemeProvider>
);

const customRender = (ui: React.ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllProviders, ...options }) as any;

const mockPanResponder = () => {
  jest
    .spyOn(PanResponder, 'create')
    .mockImplementation(
      ({
        onStartShouldSetPanResponderCapture,
        onMoveShouldSetPanResponder,
        onPanResponderGrant,
        onPanResponderMove,
        onPanResponderRelease,
      }: PanResponderCallbacks) => ({
        panHandlers: {
          onStartShouldSetResponderCapture: onStartShouldSetPanResponderCapture,
          onMoveShouldSetResponder: onMoveShouldSetPanResponder,
          onResponderGrant: onPanResponderGrant,
          onResponderMove: onPanResponderMove,
          onResponderRelease: onPanResponderRelease,
        } as GestureResponderHandlers,
      }),
    );
};

export * from '@testing-library/react-native';
export { customRender as render, mockPanResponder };
