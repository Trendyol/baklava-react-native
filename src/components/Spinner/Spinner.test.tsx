import React from 'react';
import { Animated } from 'react-native';
import { render } from '../../test-utils';
import theme from '../../theme';
import Spinner from './Spinner';
import { ReactTestInstance } from 'react-test-renderer';

describe('Spinner', () => {
  test('should render Spinner correctly', () => {
    // when
    const { toJSON } = render(<Spinner />);

    // then
    expect(toJSON()).toMatchSnapshot();
  });

  test('should render given spinner', () => {
    // when
    const { getByTestId } = render(
      <Spinner testID="spinner" color="successKey" />,
    );

    const spinnerComponent = getByTestId('spinner')
      .children[0] as ReactTestInstance;

    // then
    expect(spinnerComponent.props.children[0].props.fill).toBe(
      theme.colors.successKey,
    );
    expect(spinnerComponent.props.children[0].props.opacity).toBe(0.3);
    expect(spinnerComponent.props.children[1].props.fill).toBe(
      theme.colors.successKey,
    );
  });

  test('should render custom named color', () => {
    // when
    const { getByTestId } = render(<Spinner testID="spinner" color="pink" />);

    const spinnerComponent = getByTestId('spinner')
      .children[0] as ReactTestInstance;

    // then
    expect(spinnerComponent.props.children[0].props.fill).toBe('pink');
    expect(spinnerComponent.props.children[0].props.opacity).toBe(0.3);
    expect(spinnerComponent.props.children[1].props.fill).toBe('pink');
  });

  test('should render hex color', () => {
    // when
    const { getByTestId } = render(
      <Spinner testID="spinner" color="#5C8984" />,
    );

    const spinnerComponent = getByTestId('spinner')
      .children[0] as ReactTestInstance;

    // then
    expect(spinnerComponent.props.children[0].props.fill).toBe('#5C8984');
    expect(spinnerComponent.props.children[0].props.opacity).toBe(0.3);
    expect(spinnerComponent.props.children[1].props.fill).toBe('#5C8984');
  });

  test('should start spin animation', () => {
    const mockStart = jest.fn();
    const mockStop = jest.fn();
    const mockLoop = jest
      .fn()
      .mockReturnValue({ start: mockStart, stop: mockStop });
    jest.spyOn(Animated, 'loop').mockImplementation(mockLoop);
    // when
    render(<Spinner testID="spinner" color="successKey" />);

    // then
    expect(mockLoop).toBeCalled();
    expect(mockStart).toBeCalled();
  });

  test('should stop spin animation', () => {
    const mockStart = jest.fn();
    const mockStop = jest.fn();
    const mockLoop = jest
      .fn()
      .mockReturnValue({ start: mockStart, stop: mockStop });
    jest.spyOn(Animated, 'loop').mockImplementation(mockLoop);
    // when
    const { unmount } = render(<Spinner testID="spinner" color="successKey" />);
    unmount();

    // then
    expect(mockStop).toBeCalled();
  });
});
