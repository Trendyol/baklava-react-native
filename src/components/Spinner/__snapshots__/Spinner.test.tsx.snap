// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner should render Spinner correctly 1`] = `
<RNSVGSvgView
  accessible={false}
  align="xMidYMid"
  bbHeight={24}
  bbWidth={24}
  collapsable={false}
  fill="#FFFFFF"
  focusable={false}
  height={24}
  meetOrSlice={0}
  minX={0}
  minY={0}
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "transform": [
          {
            "rotate": "0deg",
          },
        ],
      },
      {
        "flex": 0,
        "height": 24,
        "width": 24,
      },
    ]
  }
  transform={
    [
      {
        "rotate": "0deg",
      },
    ]
  }
  vbHeight={48}
  vbWidth={48}
  width={24}
  xmlns="http://www.w3.org/2000/svg"
>
  <RNSVGGroup
    fill={
      {
        "payload": 4294967295,
        "type": 0,
      }
    }
    propList={
      [
        "fill",
      ]
    }
  >
    <RNSVGPath
      d="M48 24c0 13.255-10.745 24-24 24S0 37.255 0 24 10.745 0 24 0s24 10.745 24 24zM7.2 24c0 9.278 7.522 16.8 16.8 16.8S40.8 33.278 40.8 24 33.278 7.2 24 7.2 7.2 14.722 7.2 24z"
      fill={
        {
          "payload": 4294081050,
          "type": 0,
        }
      }
      opacity={0.3}
      propList={
        [
          "fill",
        ]
      }
    />
    <RNSVGPath
      d="M44.4 24c1.988 0 3.628-1.62 3.33-3.587A23.999 23.999 0 0027.587.27C25.62-.028 24 1.612 24 3.6s1.63 3.561 3.573 3.984a16.8 16.8 0 0112.843 12.843C40.839 22.37 42.412 24 44.4 24z"
      fill={
        {
          "payload": 4294081050,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;
